draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2652,8 +2610,276 @@ The constants used to compute y\_den are as follows:
 
 # Sample Code {#samplecode}
 
-Sample Sage {{SAGE}} code for each algorithm can be found in the draft
-repository {{hash2curve-repo}}.
+This section gives sample implementations optimized for some of the
+elliptic curves listed in {{suites}}.
+A future version of this document will include all listed curves, plus
+accompanying test vectors.
+Sample Sage {{SAGE}} code for each algorithm can also be found in the
+draft repository {{hash2curve-repo}}.
+
+## Interface and projective coordinate systems
+
+The sample code in this section uses a different interface than
+the mappings of {{mappings}}.
+Specifically, each mapping function in this section has the following
+signature:
+
+~~~
+(xn, xd, yn, nd) = map_to_curve(u)
+~~~
+
+The resulting point (x, y) is given by (xn / xd, yn / yd).
+
+The reason for this modified interface is that it enables further
+optimizations when working with points in a projective coordinate
+system.
+This is desirable, for example, when the resulting point will be
+immediately multiplied by a scalar, since most scalar multiplication
+algorithms operate on projective points.
+
+The following are two commonly used projective coordinate systems
+and the corresponding conversions:
+
+- A point (X, Y, Z) in homogeneous projective coordinates corresponds
+  to the affine point (x, y) = (X / Z, Y / Z);
+  the inverse conversion is given by (X, Y, Z) = (x, y, 1).
+  To convert (xn, xd, yn, yd) to homogeneous projective coordinates,
+  compute (X, Y, Z) = (xn * yd, yn * xd, xd * yd).
+
+- A point (X', Y', Z') in Jacobian projective coordinates corresponds
+  to the affine point (x, y) = (X' / Z'^2, Y' / Z'^3);
+  the inverse conversion is given by (X', Y', Z') = (x, y, 1).
+  To convert (xn, xd, yn, yd) to Jacobian projective coordinates,
+  compute (X', Y', Z') = (xn * xd * yd^2, yn * yd^2 * xd^3, xd * yd).
+
+## P-256 (Simplified SWU) {#map-to-p256}
+
+The following is a straight-line implementation of the Simplified SWU
+mapping for P-256 {{FIPS186-4}} as specified in {{suites-p256}}.
+
+~~~
+map_to_curve_simple_swu_p256(u)
```

Comment:
```suggestion
map_to_curve_simple_sswu_p256(u)
```

Reply:
"s" in "sswu" stands for "Simple," so I think "simple_sswu" is redundant.

In {{simple-swu}}, the function name is "map_to_curve_simple_swu", too.

Reply:
sorry it's correct.